lib/resty/core/regex.lua

```diff
@@ -520,7 +520,15 @@ local function iterate_re_gmatch(self)
         rc = 1
     end
 
-    self._pos = compiled.captures[1]
+    local cp_pos = compiled.captures[1]
```

Comment:
We should use `tonumber()` here to reduce the number of cdata objects, which are GC objects.